code/modules/clothing/modular/modular.dm

```diff
@@ -0,0 +1,427 @@
+		if(!istype(undersuit))
+			to_chat(user, "<span class='warning'>You must be wearing a marine jumpsuit to equip this.</span>")
+			return FALSE
+	return ..()
+
+/obj/item/clothing/suit/modular/attack_self(mob/user)
+	. = ..()
+	if(.)
+		return
+	if(!isturf(user.loc))
+		to_chat(user, "<span class='warning'>You cannot turn the light on while in [user.loc].</span>")
+		return
+	if(cooldowns[COOLDOWN_ARMOR_LIGHT] || !ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.wear_suit != src)
+		return
+	toggle_armor_light(user)
+	return TRUE
+
+
+/obj/item/clothing/suit/modular/item_action_slot_check(mob/user, slot)
+	if(!light_strength) // No light no ability
+		return FALSE
+	if(!ishuman(user))
+		return FALSE
+	if(slot != SLOT_WEAR_SUIT)
+		return FALSE
+	return TRUE //only give action button when armor is worn.
+
+
+/obj/item/clothing/suit/modular/attack_hand(mob/living/user)
+	if(!storage)
+		return ..()
+	if(storage.handle_attack_hand(user))
+		return ..()
+
+
+/obj/item/clothing/suit/modular/MouseDrop(over_object, src_location, over_location)
+	if(!storage)
+		return ..()
+	if(storage.handle_mousedrop(usr, over_object))
+		return ..()
+
+
+/obj/item/clothing/suit/modular/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(.)
+		return
+	if(!storage)
+		return
+
+	return storage.attackby(I, user, params)
+
+
+/obj/item/clothing/suit/modular/emp_act(severity)
+	storage?.emp_act(severity)
+	return ..()
+
+/obj/item/clothing/suit/modular/proc/can_attach(mob/living/user, obj/item/armor_module/module, silent = FALSE)
+	. = TRUE
+	if(!istype(module))
+		return FALSE
+
+	if(ismob(loc))
+		if(!silent)
+			to_chat(user, "<span class='warning'>You need to remove the armor first.</span>")
+		return FALSE
+
+	if(istype(module, /obj/item/armor_module/attachable))
+		if(length(installed_modules) >= max_modules)
+			if(!silent)
+				to_chat(user,"<span class='warning'>There are too many pieces installed already.</span>")
+			return FALSE
+		if(module in installed_modules)
+			if(!silent)
+				to_chat(user,"<span class='warning'>That module is already installed.</span>")
+			return FALSE
+
+
+	if(istype(module, /obj/item/armor_module/storage) && installed_storage)
+		if(!silent)
+			to_chat(user,"<span class='warning'>There is already an installed storage module.</span>")
+		return FALSE
+
+	if(istype(module, /obj/item/armor_module/armor/chest) && slot_chest)
+		if(!silent)
+			to_chat(user, "<span class='notice'>There is already an armor piece installed in that slot.</span>")
+		return FALSE
+	if(istype(module, /obj/item/armor_module/armor/arms) && slot_arms)
+		if(!silent)
+			to_chat(user, "<span class='notice'>There is already an armor piece installed in that slot.</span>")
+		return FALSE
+	if(istype(module, /obj/item/armor_module/armor/legs) && slot_legs)
+		if(!silent)
+			to_chat(user, "<span class='notice'>There is already an armor piece installed in that slot.</span>")
+		return FALSE
```

Comment:
Could you do a redefined proc here?

Reply:
The way I see it the modular should inform the armor where it can be equipped in, and the armor should check if that's a valid location.